mesonbuild/linkers.py

```diff
@@ -781,6 +802,13 @@ def get_allow_undefined_args(self) -> T.List[str]:
             return self._apply_prefix('--allow-shlib-undefined')
         return []
 
+    def get_win_subsystem_args(self, value: str) -> T.List[str]:
+        m = env.machines[self.for_machine]
+        if not m.is_windows() or m.is_cygwin():
+            return []
+
+        return self._apply_prefix([f'-subsystem:{value}'])
```

Comment:
This broke linking on windows with LLD in mingw mode. See above (lines 785), "Representation of LLVM's ld.lld linker. This is only the gnu-like linker, not the apple like or link.exe like linkers." - so this should use the GNU style syntax `--subsystem,windows` etc.

I'll make a PR that moves that code up from GnuDynamicLinker into GnuLikeDynamicLinkerMixin so that the same code also is used for the gnu-like uses of lld on windows.

Reply:
Good catch! I must have been under the impression that the ld compatible driver supported the link-esque subsystem syntax for some reason.